### ISSUE TITLE
Incorrect NickServ password on connect

### ISSUE THREAD

**Issue by MJ94** [lykoss/lykos#189]
The bot sends an incorrect NickServ password upon connect on a network running InsIRCd-2.0, but identifies properly the second time. We are using Python 3.2.3.

```
[2015-11-08 03:57:29+0000] <--- receive NickServ!NickServ@services.lizardirc notice (wolfbot, This nickname is registered. Please choose a different nickname, or identify via /msg NickServ identify <password>.)
[2015-11-08 03:57:29+0000] <--- receive NickServ!NickServ@services.lizardirc notice (wolfbot, Invalid password for wolfbot.)
[2015-11-08 03:57:29+0000] <--- receive NickServ!NickServ@services.lizardirc notice (wolfbot, You are now identified for wolfbot.)
[2015-11-08 03:57:29+0000] <--- receive NickServ!NickServ@services.lizardirc notice (wolfbot, 1 failed login since last login.)
```

**Comment by laneatomic** [comment]
> We are using Python 2.7.3.

Bot's written for Python 3, to start.

**Comment by Vgr255** [comment]
Please post the full contents of your `botconfig.py` file, stripping out your password. Otherwise, leave everything as-is.

**Comment by laneatomic** [comment]
Running the bot with either `--verbose` or `--debug` through the connect would help as well. If you do that, make sure you strip your password from the log before sending it.

**Comment by MJ94** [comment]
```
CHANNEL = "<redacted>"
HOST = "<redacted>"
PORT = 6697
USERNAME = ""  # for authentication, can be left blank if same as NICK
NICK = "wolfbot"
IDENT = NICK
REALNAME = NICK
CMD_CHAR = "!"
CHANGING_HOST_QUIT_MESSAGE = "Changing host"

SASL_AUTHENTICATION = False  # put account name in USERNAME ^ if different from nick
USE_SSL = True
DISABLE_DEBUG_MODE = False  # Entirely disable debug mode
IGNORE_HIDDEN_COMMANDS = True # Ignore commands sent to @#channel or +#channel
ALLOW_NOTICE_COMMANDS = False  # allow /notice #channel !command to be interpreted as a command
ALLOW_PRIVATE_NOTICE_COMMANDS = False  # allow !command's from /notice (Private)

ALLOWED_NORMAL_MODE_COMMANDS = ["revealroles", "fgame", "frevealroles"]  # debug mode commands to be allowed in normal mode

OWNERS = ("<redacted>",)  # the comma is required at the end if there is one owner
ADMINS = ("<redacted>", "<redacted>")  # glob syntax supported (wildcards)

OWNERS_ACCOUNTS = ("1owner_acc",)
ADMINS_ACCOUNTS = ("1admin_acc", "2admin_acc")

ALT_CHANNELS = ""

ALLOWED_ALT_CHANNELS_COMMANDS = []

USE_UTC = True # if True, will use the UTC time, else the local time
# {tzname} and {tzoffset} can both be used; respectively the timezone name (like UTC) and offset in the form +0000
# %Y is the year with century, %y is the year without, %m is the month, %d the day, %H hour, %M minute and %S seconds
TIMESTAMP_FORMAT = "[%Y-%m-%d %H:%M:%S{tzoffset}]"

# If enabled and DEV_CHANNEL is set, the bot will post tracebacks there.
PASTEBIN_ERRORS = False

# Channel(s) where errors will be sent to. Can be prefixed with '+' or '@' for sending to only voices or ops.
DEV_CHANNEL = ""
```

**Comment by FastLizard4** [comment]
I looked through the bot's output myself, and I can confirm that the problem is that the bot is sending `PASS wolfbot:password` during registration, and then later also sends a `PRIVMSG NickServ :IDENTIFY wolfbot password`.  In any case, the "real" cause of this is that the network in question expects the username and password of the NickServ account, if specified with PASS during registration, to be separated with a space, not a colon like on freenode.

**Comment by nyuszika7h** [comment]
Looks like oyoyo's code has to be modified for this. >_>

**Comment by laneatomic** [comment]
What services package are you using for SASL? Specifying a PASS string with a space in the password doesn't look like it should be possible, according to RFC 2812.

>3.1.1 Password message
>
>   Command: PASS
>   Parameters: \<password\>
>
>   The PASS command is used to set a 'connection password'.  The
>   optional password can and MUST be set before any attempt to register
>   the connection is made.  Currently this requires that user send a
>   PASS command before sending the NICK/USER combination.
>
>   Numeric Replies:
>
>           ERR_NEEDMOREPARAMS              ERR_ALREADYREGISTRED
>
>   Example:
>
>           PASS secretpasswordhere

**Comment by FastLizard4** [comment]
We're using Atheme 7.2.5 with InspIRCd 2.0.20.  Even if it isn't in the RFC spec, it does seem to work - it's how I identify on connect.  It's worth noting that the forwarding of the `PASS` command to NickServ is done by an InspIRCd module in this case, which simply translates the argument to `PASS` to the argument to a `PRIVMSG NickServ :IDENTIFY` command.

**Comment by skizzerz** [comment]
Fixed in #190 -- you'll need to add SERVER_PASS to your botconfig.py -- check botconfig.py.example for an example of how to format it.
